crates/autopilot/src/domain/settlement/trade/math.rs

```diff
@@ -441,6 +410,27 @@ impl Trade {
     }
 }
 
+/// Fee per trade in a solution. These fees are taken for the execution of the
+/// trade.
+#[derive(Debug, Clone)]
+pub struct FeeBreakdown {
+    /// Fee reported by solvers, representing difference between UCP and custom
+    /// prices.
+    pub executed: eth::TokenAmount,
+    /// Breakdown of protocol fees.
+    pub protocol: Vec<ExecutedProtocolFee>,
+    /// Surplus token in which all the fees were taken.
+    pub token: eth::TokenAddress,
+}
```

Comment:
It would be helpful to include details from your [comment](https://github.com/cowprotocol/services/pull/3184/files#r1898434838) somewhere here.